doc/use_cases.md

```diff
@@ -0,0 +1,49 @@
+
+For this use case we would have a smaller set of posteriors as the requirements are very strict. Initial set comes from https://github.com/stan-dev/stat_comp_benchmarks. This set can be gradually expanded.
+
+In this use case we assume an algorithm should work well for all posteriors in the strict testing set.
+
+This kind of strict testing could be run every time before new release and every time after any algorithm changes in a development branch.
+
+## Code speed testing
+
+If the above strict test is passed, the same test can be used for timing comparisons.
+
+Even if the algorithm doesn't change, various changes in, for example, Stan math library, autodiff, parallelization implementation, and compiler options can affect the computation speed. The small test set can be used to test code speed by running earlier and new version in the same computer. See example in https://github.com/stan-dev/performance-tests-cmdstan
+
+## Efficiency comparisons of inference algorithms with asymptotically decreasing bias and variance
+
+Assuming we have different inference algorithms, small variants of some specific algorithm, for example, having different warm-up adaptations in dynamic HMC, and the implementations of these have passed the above strict checking. We are interested in testing efficiency of these algorithms with more complex posteriors. Now to make the set of more interesting models have enough variety, we assume that careful use dynamic HMC, convergence diagnostics, and SBC will provide the reference posterior. We don't claim these reference posteriors would be 100% correct, and if your new algorithm produces different result there is reason to check which result is more likely to closer to true posterior. While in efficiency comparisons we assume already the algorithms are working (ie they have passed the previous more strict step), it is still useful to make some checks that produced  posterior is close to the reference posterior. The main interest in this use case is however in inference diagnostics and efficiency measures. Inference diagnostics should detect if the tested inference algorithm is not able to produce accurate results. An inference algorithm can be useful if it works fast for some set of models and it can be diagnosed when it doesn't work. Useful efficiency measures are effective sample size divided by the number of log-density and gradient evaluations and effective sample size per second. The first one is useful when comparing high-level algorithms as it doesn't care how well log-density and gradient evaluations are optimized and the second one can be useful when comparing algorithms where the choice of algorithm can affect the computational cost of log-density and gradient evaluation (e.g. use of Laplace integration over the latent values)-
+
+The bigger set will be filled with models from https://github.com/stan-dev/example-models 
```

Comment:
Bigger set in comparison to what smaller set?

Reply:
Small set is initially filled from https://github.com/stan-dev/stat_comp_benchmarks

Reply:
Great. Should be made explicit then.

Reply:
We already state explicitly " a smaller set of posteriors as the requirements are very strict. Initial set comes from https://github.com/stan-dev/stat_comp_benchmarks." Do you have suggestions how to improve?

Reply:
Sorry, must have overlooked that.

Reply:
It is far away, so should we repeat it?

Reply:
It could make sense given that "bigger" is at least locally out of context.

Reply:
I removed this sentence. We want the document not to describe what we are going to do too much since then we will need to change it.